src/components/Navbar/client.tsx

```diff
@@ -11,91 +12,116 @@ import {
   NavigationMenuList,
   NavigationMenuTrigger,
   NavigationMenuViewport,
+  navigationMenuTriggerStyle,
 } from "@/components/ui/navigation-menu";
 import { Button } from "@/components/ui/button";
 import { Separator } from "@/components/ui/separator";
+import { signoutUser } from "@/lib/User/actions";
 
 import { getIcon } from "@/data/static";
 import { Technologies, AccountLinks } from "@/data/navigation";
 const technologies = Technologies;
 const accountLinks = AccountLinks;
 
-// Generate the Account navigation menu
-const Account = () => {
-  return (
-    <ul className="grid w-[400px] md:w-[200px] md:grid-cols-1 gap-row-10">
-      {accountLinks.map((link) => (
-        <li key={link.text}>
-          {link.style === "button" ? (
-            <Button className="w-full block p-3 px-6">
-              <Link href={link.href}>{link.text}</Link>
-            </Button>
-          ) : (
-            <Link
-              href={link.href}
-              className="px-6 p-2 block text-slate-500 border-b-[1px] hover:text-slate-900 hover:bg-slate-100 focus:text-slate-900 focus:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:ring-slate-900 focus-visible:ring-offset-slate-100 transition-colors"
-            >
-              {link.text}
-            </Link>
-          )}
-        </li>
-      ))}
-    </ul>
-  );
-};
+export const NavbarItems = ({ loggedIn }: { loggedIn: boolean }) => {
+  const [userLoggedIn, setUserLoggedIn] = useState(loggedIn);
```

Comment:
We shouldn't need to store this in state anywhere.

Reply:
I think it's about `optimistic updates`, but yeah. It's better to not have state if possible

Reply:
Right! That makes sense lol

Reply:
Yeah, we shouldn't rely on that. Any usage of state requires us to opt-in to "use client" which isn't good practice.

Reply:
> Yeah, we shouldn't rely on that. Any usage of state requires us to opt-in to "use client" which isn't good practice.

Oh wow. I didn't know that. I would love to get more insight into this and understand why it isn't good practice. Thanks for letting me know!

Reply:
Check out https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns to learn more about server vs client components.